File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        result = []
        
        hashmap = {} #  number -> index
        
        for i, el in enumerate(nums):
            if target - el in hashmap:
                result.append(hashmap[target-el])
                result.append(i)
                return result
            
            hashmap[el] = i
            
        return result
            

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int, int> mp; // value -> indexes
        vector<int> result;
        
        for(int i=0; i<nums.size(); i++){
            if(mp.find(target-nums[i]) != mp.end()){
                result.push_back(mp[target-nums[i]]);
                result.push_back(i);
                return result;
            }
            
            mp[nums[i]] = i;
            
        }
        
        return result;
    }
};